contrib/unbound_munin_

```diff
@@ -121,12 +120,24 @@ get_value ( ) {
 	fi
 }
 
+# Update list of seen query types etc to seentags file. This is run while
+# holding the lock, after the state file is updated.
+update_seentags() {
+    tmplist="$(cat ${seentags} 2> /dev/null)
+num.query.type.A
+num.query.class.IN
+num.query.opcode.QUERY
+num.answer.rcode.NOERROR
+"
+    (echo "${tmplist}"; grep ^num ${state} | sed -e 's/=.*//') | sort -u > ${seentags}
+}
+
```

Comment:
OK, so here is the core of the change. If I understand correctly you want munin to keep plotting tags that unbound was reporting on before the restart (with 0 as the value if the tag is no longer reported by unbound; based on get_value()).

But why do you always set the above 4?

Reply:
Yes, you understood it correctly.

Those 4 tags are needed, as later on code has "`for nm in grep "^num.query.type" $seentags; do`" (and num.query,class, opcode and rcode). This makes it sure that there are at least one tag for each graph, needed by Munin. Graph without tag is not allowed.